activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/TaskPermissionsHelper.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.query.rest;
+
+import org.activiti.api.runtime.shared.security.SecurityManager;
+import org.activiti.cloud.api.task.model.QueryCloudTask.TaskPermissions;
+import org.activiti.cloud.services.query.model.QTaskEntity;
+import org.activiti.cloud.services.query.model.TaskEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskPermissionsHelper {
+    
+    private final SecurityManager securityManager;
+
+    private final TaskControllerHelper taskControllerHelper;
+
+    public TaskPermissionsHelper(SecurityManager securityManager, TaskControllerHelper taskControllerHelper) {
+        this.securityManager = securityManager;
+        this.taskControllerHelper = taskControllerHelper;
+    }
+
+    public void setCurrentUserTaskPermissions(TaskEntity task) {
+        String userId = securityManager.getAuthenticatedUserId();
+        if(userId!=null) {
+            List<String> userGroups = securityManager.getAuthenticatedUserGroups();
+            List<TaskPermissions> permissions = new ArrayList<>();
+            if(!canUserViewTask(task)) return;
+            
+            permissions.add(TaskPermissions.VIEW);
+            
+            if(canUserClaimTask(task, userId, userGroups)) {
+                permissions.add(TaskPermissions.CLAIM);
+            }
+            
+            if(isUserAssignee(task, userId)) {
+                permissions.add(TaskPermissions.RELEASE);
```

Comment:
`RELEASE` should be only possible if the task is assigned to current user and the user is a candidate or a member of a candidate group

Reply:
I think is not necessary to check if the user is also a candidate of a member if he is the assignee. I also found this in rb [TaskRuntimeImpl.java#L475-L485](https://github.com/Activiti/Activiti/blob/ecec52d763b4ce0fe0362a0f3996b28bd877c318/activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java#L475-L485).
 What do you think?

Reply:
My understanding to `RELEASE` is to release back the task to the queue. So if there is no candidates or candidate groups where can this task release back to?

Reply:
I'm not sure this scenario is possible. Can we replicate it? I guess that If you are the assignee, you were a candidate user or member of a candidate group.  I think the permissions should follow the behaviour of the rb API.

Reply:
`RELEASE` is opposite to `CLAIM` so if the task wasn't claimed the task it shouldn't be possible to release it.